src/util.c

```diff
@@ -151,6 +151,24 @@ char *progname(char *arg0)
        return prognm;
 }
 
+/**
+ * basenm - Our custom basename implementation
+ * @path: The path to parse
+ *
+ * Returns either the parsed basename, or
+ * a copy of the given path
+*/
+char *basenm(const char *path) {
+	char *basename;
+
+	basename = rindex(path, '/');
+	if (!basename)
+		return (char *) path;
+
+	basename++;
+	return basename;
+}
+
 char *str(char *fmt, ...)
```

Comment:
The comment is very confusing, the function never returns a copy.  Just simplify it as:

```C
/* basename(3) replacement that does not modify its argument */
```

Please follow the coding style of the project, or at least, surrounding code.  The opening brace of the function goes on the next line.

Also, I'm not overly happy with the conversion from const to non-const here.  But making the function return a `const char *` probably causes a ripple effect on more of the code?

Reply:
Will push the changes in a few seconds.

Yes, sadly having it return `const char *` will cause a ripple effect, though currently only 2 places in the code base come to mind (as of now), `src/mdadm.c:62` and `src/plugin.c:92` where the return value of `basenm()` is assigned to `char *` in both cases.

`src/mdadm.c:62` could be made `const char *` without problem
`src/plugin.c:92` not so much, as it's reassigned and from the looks of it, not so trivial to make it `const char *` as well

Reply:
OK. Makes sense. I'll probably have a look at it myself later on. 

Thank you for taking the time to report and fix this!